pkg/util/namespace/namespace.go

```diff
@@ -3,23 +3,70 @@ package namespace
 // Copyright (c) Microsoft Corporation.
 // Licensed under the Apache License 2.0.
 
-import (
-	"strings"
-)
-
-// IsOpenShift returns true if ns is an openshift managed namespace, including the default namespace
-func IsOpenShift(ns string) bool {
-	return ns == "" ||
-		ns == "default" ||
-		ns == "openshift" ||
-		strings.HasPrefix(ns, "kube-") ||
-		strings.HasPrefix(ns, "openshift-")
-}
-
-// IsOpenShiftSystemNamespace returns true if ns is an openshift managed namespace, without the default namespace
-func IsOpenShiftSystemNamespace(ns string) bool {
-	return ns == "" ||
-		ns == "openshift" ||
-		strings.HasPrefix(ns, "kube-") ||
-		strings.HasPrefix(ns, "openshift-")
+// FilteredOpenShiftNamespace returns true if ns is a namespace in the defined hardcoded map
+func FilteredOpenShiftNamespace(ns string) bool {
+	filteredNamespaces := map[string]struct{}{
+		"":                                                 {},
```

Comment:
As mentioned in other comments in this PR, please would you separate the ARO and OCP namespaces so our future selves can more easily curate the lists. Also, please add a comment about the empty "" namespace so it's clear why it's necessary.

Separately, when compared to the previous functions definition, I would say this list is also missing the `kube-system` namespace.

Reply:
@darthhexx - Thanks for your review feedback.  I hard-coded the namespaces mainly based on the [configMap](https://github.com/openshift/managed-cluster-config/blob/ade1697fd61ff033f8fe24134df6bba3a390a000/deploy/osd-managed-resources/ocp-namespaces.ConfigMap.yaml) and it doesn't contain `kube-system` namespace. 

Do we really need to add kube-system related namespace? If so, what particular `kube-system` namespace we should hard-code? The previous functions just check namespaces by patterns `strings.HasPrefix(ns, "kube-")` but I wasn't aware in the ADO task that the kube-system needs to be included ?

Reply:
I haven't checked all of the places that the function is called, but the original functions would have matched on `kube-system`, which we do interact with. I was merely highlighting the fact that the new function is not only being stricter about openshift namespaces but also dropping kube-* namespace support. Is this intended?

Reply:
@m1kola - As David mentioned above, I also wanted to get clarified that if the `kube-*` namespaces should be hardcoded or we only need to filter `openshift-` namespaces in the new hard-coded function as intended?